native/core/src/execution/shuffle/row.rs

```diff
@@ -271,6 +423,51 @@ impl SparkUnsafeRow {
             *word_offset = word & !mask;
         }
     }
+
+    #[inline]
+    pub fn set_null_at(&mut self, index: usize) {
+        unsafe {
+            let mask: i64 = 1i64 << (index & 0x3f);
+            let word_offset = (self.row_addr + (((index >> 6) as i64) << 3)) as *mut i64;
+            let word: i64 = *word_offset;
+            *word_offset = word | mask;
+        }
+    }
+
+    #[inline]
+    pub fn set_boolean(&mut self, index: usize, value: bool) {
```

Comment:
These `set` methods could potentially be split out into an independent PR if we are unsure about merging this one

Reply:
I can create a PR but I don't see the usefulness of doing so. It's looking like creating an `UnsafeRow` in native is a slower approach in general, so best avoided.